samples/apply-hillshade-renderer-to-raster/src/main/java/com/esri/arcgismaps/sample/applyhillshaderenderertoraster/screens/ApplyHillshadeRendererToRasterScreen.kt

```diff
@@ -0,0 +1,254 @@
+
+    // Set up the bottom sheet controls
+    val controlsBottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+    var isBottomSheetVisible by remember { mutableStateOf(false) }
+    LaunchedEffect(isBottomSheetVisible) {
+        if (isBottomSheetVisible) controlsBottomSheetState.show()
+        else controlsBottomSheetState.hide()
+    }
+
+    // Collect latest UI states
+    val altitude by remember { mutableDoubleStateOf(mapViewModel.currentAltitude) }
+    val azimuth by remember { mutableDoubleStateOf(mapViewModel.currentAzimuth) }
+    val slopeType by remember { mutableStateOf(mapViewModel.currentSlopeType) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+            ) {
+                MapView(
+                    arcGISMap = mapViewModel.arcGISMap,
+                    modifier = Modifier.fillMaxSize()
+                )
+                if (isBottomSheetVisible) {
+                    ModalBottomSheet(
+                        modifier = Modifier.wrapContentHeight(),
+                        sheetState = controlsBottomSheetState,
+                        onDismissRequest = { isBottomSheetVisible = false }
+                    ) {
+                        HillshadeRendererOptions(
+                            currentAltitude = altitude,
+                            currentAzimuth = azimuth,
+                            currentSlopeType = slopeType,
+                            onApplyRenderer = mapViewModel::updateRenderer
+                        )
+                    }
+                }
+            }
+        },
+        floatingActionButton = {
+            if (!isBottomSheetVisible) {
+                FloatingActionButton(
+                    modifier = Modifier.padding(bottom = 36.dp, end = 12.dp),
+                    onClick = { isBottomSheetVisible = true }
+                ) { Icon(Icons.Filled.Settings, contentDescription = "Hillshade renderer options") }
+            }
+        }
+    )
+}
+
+@Composable
+fun HillshadeRendererOptions(
+    currentAltitude: Double,
+    currentAzimuth: Double,
+    currentSlopeType: SlopeType,
+    onApplyRenderer: (Double, Double, SlopeType) -> Unit
+) {
+    var altitude by remember { mutableIntStateOf(currentAltitude.roundToInt()) }
+    var azimuth by remember { mutableIntStateOf(currentAzimuth.roundToInt()) }
+    var selectedSlopeType by remember { mutableStateOf(currentSlopeType) }
+    val slopeTypes = listOf(
+        SlopeType.None, SlopeType.Degree, SlopeType.PercentRise, SlopeType.Scaled
+    )
+
+    // Apply renderer when any one of these values are updated
+    LaunchedEffect(altitude, azimuth, selectedSlopeType) {
+        onApplyRenderer(altitude.toDouble(), azimuth.toDouble(), selectedSlopeType)
+    }
```

Comment:
I am not sure this is necessary as this could up called directly when the slider changes.